engine/api/event_v2/event_entity.go

```diff
@@ -0,0 +1,65 @@
+package event_v2
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ovh/cds/engine/cache"
+	"github.com/ovh/cds/sdk"
+)
+
+func PublishEntityDeleteEvent(ctx context.Context, store cache.Store, vcsName, repoName string, ent sdk.Entity, u *sdk.AuthentifiedUser) {
+	bts, _ := json.Marshal(ent)
+	e := sdk.EventV2{
+		ID:         sdk.UUID(),
+		ProjectKey: ent.ProjectKey,
+		VCSName:    vcsName,
+		Repository: repoName,
+		Entity:     ent.Name,
+		Type:       sdk.EventEntityDeleted,
+		Payload:    bts,
+	}
+	if u != nil {
```

Comment:
if the user can be nil if some case, we should never set it.

event data must be reliable